Source/GameEngine/GameEngine.cs

```diff
@@ -0,0 +1,252 @@
+            {
+                foreach (GamePiece gamePiece in player.GamePieces)
+                {
+                    if (gamePiece.position.BoardPosition == currentPosition)
+                    {
+                        var gamePieceOnPosition = gamePiece;
+
+                        if (gamePieceOnPosition != ourGamePiece)
+                        {
+                            PushGamePiece(gamePieceOnPosition, player);
+                        }
+                    }
+                }
+            }
+        }
+
+        public void PushGamePiece(GamePiece gamePiece, Player player)
+        {
+            gamePiece.position.positionType = PositionType.StartingPosition;
+
+            if (player.Color == Color.Red)
+            {
+                gamePiece.position.BoardPosition = 1;
+            }
+            else if (player.Color == Color.Blue)
+            {
+                gamePiece.position.BoardPosition = 11;
+            }
+            else if (player.Color == Color.Yellow)
+            {
+                gamePiece.position.BoardPosition = 21;
+            }
+            else if (player.Color == Color.Green)
+            {
+                gamePiece.position.BoardPosition = 31;
+            }
+        }
+
+        public GamePiece StepCounter(GamePiece gamePiece, int dice)
+        {
+            if (gamePiece.StepCounter >= 40)
+            {
+                gamePiece.position.positionType = PositionType.InnerPath;
+
+                var innerPathSteps = gamePiece.StepCounter - 40;
+
+                gamePiece.StepCounter = gamePiece.StepCounter + innerPathSteps;
+                gamePiece.position.BoardPosition = gamePiece.position.BoardPosition = innerPathSteps;
+
+                if (gamePiece.StepCounter > 5)
+                {
+                    gamePiece.StepCounter -= 5;
+                }
+                else if (gamePiece.StepCounter == 5)
+                {
+                    players[0].Score += 1;
+                }
+                else
+                {
+                    gamePiece.StepCounter += dice;
+                }
+
+            }
+            return gamePiece;
+        }
+
+
+        public void ChooseStartingPlayer()
+        {
+            var dice = 0;
+            while (players.Count != playerOrder.Count)
+            {
+                foreach (Player player in players)
+                {
+                    dice = RollDice();
+                    player.StartingDice = dice;
+                    Console.WriteLine($"{player.Name} rolled {dice}.");
+                }
+
+                playerOrder = players.OrderByDescending(p => p.StartingDice).ToList();
+                Console.WriteLine($"{playerOrder[0].Name} is the starting player!");
+            }
+        }
+
+        public void MoveGamePieceToBoard(Player player, GamePiece gp, int dice)
+        {
+            gp.position.positionType = PositionType.OuterPath;
+            MoveGamePiece(gp, dice);
+
+        }
+
+
+    }
+}
```

Comment:
Väldigt tydliga metoder!! : ) enkelt att se på namnet vad de ska göra och koden gör det lätt att se också att de gör det som "says on the tin" så att säga!! 👍